### ISSUE TITLE
warning: unknown attribute 'optimize' ignored [-Wunknown-attributes]

### ISSUE THREAD

**Issue by nlohmann** [eranpeer/FakeIt#238]
With Clang (Apple clang version 12.0.0 (clang-1200.0.32.29)), I get the following warning compiling FakeIt:

```
../external/fakeit/fakeit.hpp:5291:7: warning: unknown attribute 'optimize' ignored [-Wunknown-attributes]
    [[gnu::optimize("no-devirtualize")]]
      ^
```

This seems to be a GNU extension. I would expect CMake (or the preprocessor) to find out which compiler I am using and only use those attributes that are supported.

**Comment by FranckRJ** [comment]
Should be fixed on 2.0.8, which version are you using ?

Le jeu. 15 avr. 2021 à 09:20, Niels Lohmann ***@***.***> a
écrit :

**Comment by nlohmann** [comment]
Indeed the warning is gone with version 2.0.8. Sorry for not trying this earlier.
